jdtls.ext/com.microsoft.java.maven.plugin/src/main/java/com/microsoft/java/maven/AddDependencyHandler.java

```
@@ -0,0 +1,266 @@
+import org.eclipse.jdt.ls.core.internal.JDTUtils;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+import org.eclipse.lsp4j.WorkspaceEdit;
+import org.xml.sax.Attributes;
+import org.xml.sax.Locator;
+import org.xml.sax.SAXException;
+import org.xml.sax.XMLReader;
+import org.xml.sax.helpers.DefaultHandler;
+
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AddDependencyHandler {
+
+    public static class AddDependencyParams {
+
+        final String fullClassName;
+        final String artifactInfo;
+        final String uri;
+        final int line;
+        final int character;
+        final int length;
+
+        public AddDependencyParams(String fullClassName, String artifactInfo, 
+                String uri, int line, int character, int length) {
+            this.fullClassName = fullClassName;
+            this.artifactInfo = artifactInfo;
+            this.uri = uri;
+            this.line = line;
+            this.character = character;
+            this.length = length;
+        }
+    }
+
+    private static WorkspaceEdit importEdit(AddDependencyParams params, 
+            ICompilationUnit unit, CompilationUnit astRoot) {
+        // import the new class
+        final CompilationUnitChange cuChange = new CompilationUnitChange("", unit);
+        final ImportRewrite importRewrite = CodeStyleConfiguration.createImportRewrite(astRoot, true);
+        importRewrite.addImport(params.fullClassName);
+        try {
+            cuChange.setEdit(importRewrite.rewriteImports(null));
+            return ChangeUtil.convertToWorkspaceEdit(cuChange);
+        } catch (CoreException e) {
+            return new WorkspaceEdit();
```

reason to swallow the exception? suggest to at least do some logging in case of future diagnose

---

to do